### PR TITLE
feat: POST /vc-api/exchanges "409 Conflict" response when collision detected

### PR DESCRIPTION
**Pull request by artursudnik**
This PR adds logic to the `POST /vc-api/exchanges` endpoint to respond with "409 Conflict" response when submitting exchangeId already used.

TODO:

- [x] adjusting tests failing after update

### CODE REVIEW

File: apps/vc-api/src/vc-api/exchanges/exchange.service.spec.ts [hunk 1]
```diff
@@ -60,13 +60,17 @@ describe('ExchangeService', () => {
...
-  }));
+
+  const repositoryMockFactory = jest.fn(() => {
+    let exchange: ExchangeEntity;
+
+    return {
+      findOne: jest.fn(() => exchange),
+      save: jest.fn((entity: ExchangeEntity) => {
+        exchange = entity;
+      })
```
Comment (Harasz) [inline apps/vc-api/src/vc-api/exchanges/exchange.service.spec.ts:71]: I know it's outside of the task scope, but i think `save` should actually return saved value, to met `typeorm` behavior

File: apps/vc-api/src/vc-api/exchanges/exchange.service.ts [hunk 2]
```diff
@@ -44,6 +45,10 @@ export class ExchangeService {
   ) {}
 
   public async createExchange(exchangeDefinitionDto: ExchangeDefinitionDto) {
+    if (!isNil(await this.exchangeRepository.findOne(exchangeDefinitionDto.exchangeId))) {
```
Comment (Harasz) [inline apps/vc-api/src/vc-api/exchanges/exchange.service.ts:48]:
```suggestion
    if (!await this.exchangeRepository.findOne(exchangeDefinitionDto.exchangeId)) {
```

Why we can't simplify it?

File: apps/vc-api/src/vc-api/exchanges/exchange.service.ts [hunk 3]
```diff
@@ -44,6 +44,10 @@ export class ExchangeService {
   ) {}
 
   public async createExchange(exchangeDefinitionDto: ExchangeDefinitionDto) {
+    if (!!(await this.exchangeRepository.findOne(exchangeDefinitionDto.exchangeId))) {
```
Comment (Harasz) [inline apps/vc-api/src/vc-api/exchanges/exchange.service.ts:47]:
```suggestion
    if (await this.exchangeRepository.findOne(exchangeDefinitionDto.exchangeId)
    )) {
```
Will this work?

Comment (artursudnik) [reply apps/vc-api/src/vc-api/exchanges/exchange.service.ts:47]: Right! It's too hot day today :D

**Review by jrhender — APPROVED** [review]
Looks good to me!

**Pull request merged**